### DISCUSSION TITLE
Having a strange issue

### DISCUSSION THREAD

**Discussion by jasonhutton-sd** [gilbarbara/react-joyride#808] [Q&A]
When I step through steps and I get to an input field, I cannot edit or even focus on that field to type. I am using MUI for inputs.
To add to the issue, when I click away to another window and back to where joyride is running I can then edit the input field.

Please help.

Thanks

**Comment by jasonhutton-sd** [comment]
I did some more testing. This behavior seems to be the same even without using MUI. Seems like the "Next" button is constantly wanting to take the focus. @gilbarbara any ideas here? This is super frustrating and the issue did not seem to be present in an older version of react-joyride (2.2.1) and I am using version 2.4.

**Reply by gilbarbara** [reply]
hey @jasonhutton-sd 

No idea...
It's impossible to know what's happening without viewing the code.
Please post a codesandbox with a minimal reproduction example.

**Reply by gilbarbara** [reply]
About the tab focus, while the step's tooltip is active the focus is kept within.

**Reply by jasonhutton-sd** [reply]
Kept within Joyride? I assume that would not let you type into an input?

**Reply by gilbarbara** [reply]
Only the **tab focus**, all other keystrokes aren't captured.
Anyway, without an example, there's nothing else I can do.

**Reply by jasonhutton-sd** [reply]
@gilbarbara I hope this helps
https://codesandbox.io/s/wandering-shape-ropcwp?file=/src/App.js

**Comment by gilbarbara** [comment]
@jasonhutton-sd 

You should set [spotlightClicks](https://docs.react-joyride.com/props) in the step (or as a prop in the component).
Keyboard navigation won't work, but you can set the input focus programmatically if needed.
https://codesandbox.io/s/react-joyride-input-focus-demo-h0v17q

**Reply by jasonhutton-sd** [reply]
Something else going on. The example above works. But in my project, as soon as focus is set to the input, focus is redirected back to the tooltip button.
I am guessing the example I provided was too simple? Not sure where to go from here and thanks for the help.

**Reply by gilbarbara** [reply]
Increase the timeout, Idk.
Good luck
